pennylane/operation.py

```diff
@@ -2261,6 +2259,9 @@ def prune(self):
         Returns:
             ~.Observable: the pruned tensor product of observables
         """
+        if qml.QueuingManager.recording():
+            qml.QueuingManager.remove(self)
```

Comment:
should this have been there before? why is it being added now? I suppose we're assuming that if a user calls `some_tensor.prune()`, they intend only for the pruned result to be queued

Reply:
Probably doesn't matter that much since Tensor's don't actually make it into the circuit anyway. But I think this is another example of an operator being replaced by a new version of it.